docs/application/dotnet/guides/app-management/event.md

```diff
@@ -0,0 +1,291 @@
+
+- libslp-location
+
+  - location
+
+    | Event name | Event data Key | Event data Value | Condition | Notes |
+    |-------|-------|--------|--------|--------|
+    | `SystemEvents.LocatingEnableState.EventName` | `SystemEvents.LocatingEnableState.StateKey`   | - `SystemEvents.LocatingEnableState.StateValueDisabled`:  Location disabled <br>-  `SystemEvents.LocatingEnableState.StateValueEnabled`: Location enabled | When the `location-enable_state` has been  changed, for example, by the user toggling the location setting in the  settings menu or quick panel. | You can get the  current value with the `Location.LocatorHelper.IsEnabledType` property. |
+    | `SystemEvents.GpsEnableState.EventName`      | `SystemEvents.GpsEnableState.StateKey`        | - `SystemEvents.GpsEnableState.StateValueDisabled`:  GPS disabled  <br>- `SystemEvents.GpsEnableState.StateValueEnabled`: GPS enabled | When the `gps-enable_state` has changed.   | You can get the  current value with the `Location.LocatorHelper.IsEnabledType` property. |
+    | `SystemEvents.NpsEnableState.EventName`      | `SystemEvents.NpsEnableState.StateKey`        | - `SystemEvents.NpsEnableState.StateValueDisabled`:  NPS disabled <br>-  `SystemEvents.NpsEnableState.StateValueEnabled`: NPS enabled | When the NPS  setting has been changed, for example, by the user toggling the location  settings. | You can get the  current value with the `Location.LocatorHelper.IsEnabledType` property. |
+
+- msg-service
+
+  - message
+
+    | Event name | Event data Key | Event data Value | Condition | Notes |
+    |-------|-------|--------|--------|--------|
+    | `SystemEvents.IncomingMsg.EventName`          | `SystemEvents.IncomingMsg.TypeKey`                | - `SystemEvents.IncomingMsg.TypeValueSms`:  SMS-type message <br>-  `SystemEvents.IncomingMsg.TypeValueMms`: MMS-type message <br>-  `SystemEvents.IncomingMsg.TypeValuePush`: Push-type message <br>-  `SystemEvents.IncomingMsg.TypeValueCb`: Cb-type message | When an SMS, MMS, push, or CB message has been received. |       -                                   |
+    | `SystemEvents.IncomingMsg.EventName`          | `SystemEvents.IncomingMsg.IdKey`                  | `msg_id`: Message ID of the received message (string of the unsigned `int` type value) | -                                         |           -                               |
+    | `SystemEvents.OutgoingMsg.EventName`          | `SystemEvents.OutgoingMsg.TypeKey`                | - `SystemEvents.OutgoingMsg.TypeValueSms`:  SMS-type message <br>-  `SystemEvents.OutgoingMsg.TypeValueMms`: MMS-type message | When an SMS or MMS message has been sent. |       -                                   |
+    | `SystemEvents.OutgoingMsg.EventName`          | `SystemEvents.OutgoingMsg.IdKey`                  | `msg_id`: Message ID of the received message (string of the unsigned `int` type value) | -                                         |           -                               |
```

Comment:
Just a question. It looks like this row is talking about "OutgoingMsg". Is the `msg_id` really the Message ID of the **received** message?

Reply:
It's my fault. It means the **Sent** message